core/container.go

```diff
@@ -363,8 +379,8 @@ func (s *containerSchema) Resolvers() router.Resolvers {
 			"from":                 router.ToResolver(s.from),
 			"rootfs":               router.ToResolver(s.rootfs),
 			"directory":            router.ErrResolver(ErrNotImplementedYet),
-			"user":                 router.ErrResolver(ErrNotImplementedYet),
-			"withUser":             router.ErrResolver(ErrNotImplementedYet),
+			"user":                 router.ToResolver(s.user),
```

Comment:
We should make follow ups for `withGroup` and `withSupplementaryGroup(s)`. Not blocking at all for this PR of course or even that high priority, just something we'll want eventually.

I just noticed that buildkit LLB is missing those though. So we'd have to either contribute upstream (ideal) or implement it via the shim (less ideal but fine if needed).

Reply:
This might actually work as-is, since it looks like the OCI config spec has it embedded in the `User` field like `user:group`:

https://github.com/opencontainers/image-spec/blob/main/config.md#properties

Not sure about supplementary groups though.

I'll add a test for this along with setting numeric UID/GID.

Reply:
You're right, good catch, this also just reignited a memory that I at one point had to deal with all this in old dagger, so I apparently knew about the `user:group` thing at one point and forgot: https://github.com/dagger/dagger/pull/2542 😂

Supplementary groups are fine to be missing for now I think; the use cases become much more obscure with them (`wheel` and various ones for devices like `gpu` and `kvm` are what I'm aware of, but that's just not important for us right now).